adaptdl/adaptdl/torch/gradient_noise_scale.py

```diff
@@ -114,32 +71,35 @@ def __init__(self, adp, optimizer, mp_scaler=None, num_replicas=None,
                 param.register_hook(
                     functools.partial(self._backward_hook, idx, param))
         self._callback_queued = False
-
-        if patch_optimizer:
-            self.patch_optimizer()
-
         self._smoothing = 0.999
 
-        self._reset_accumulation()
+    @property
+    def _state(self):
+        return self._optimizer.state["gns"]
 
     def _reset_accumulation(self):
+        """reset accumulation calculations and gradients."""
         self._orig_optimizer_zero_grad()
         self._local_sqr = None
         self._accum_count = 0
 
-    @property
-    def _state(self):
-        return self._optimizer.state["adascale"]
-
     @property
     def accum_scale(self):
         return self._accum_scale
 
+    @property
+    def accum_count(self):
+        return self._accum_count
+
     def set_accum_scale(self, accum_scale):
         if not np.isclose(self._accum_scale, accum_scale):
             self._reset_accumulation()
             self._accum_scale = accum_scale
 
+    @property
+    def raw_sqr_avg(self):
+        return self._state["sqr_avg"]
```

Comment:
Recommend
```
view = self._state["sqr_avg"].view()
view.flags.writeable = False
return view
```
prevents `self._state["sqr_avg"]` from being accidentally mutated by the caller of this method. Same with `raw_var_avg`.